web/services/createProjectsService.php

```diff
@@ -208,11 +209,37 @@
 
 
     if (count($createProjects) >= 1)
-        if (ProjectsFacade::CreateProjects($createProjects) == -1)
-            $string = "<return service='createProjects'><error id='1'>There was some error while creating the projects</error></return>";
+        if(count($createProjects) == 1){
+            $operationResult = ProjectsFacade::CreateProject($createProjects[0]);
+            if (!$operationResult->isSuccessful)
+                $string = "<return service='createProjects'><error id='1'>". $operationResult->message . "</error></return>";
+        }
+        else {
+            //leaving the possibility to create multiple projects in the future
+            $operationResults = ProjectsFacade::CreateProjects($createProjects);
+            $string = "<return service='createProjects'>";
+            $errors = array_filter($operationResults, function ($item) {
+                return ($item->isSuccessful = false);
+            });
+            if($errors){
+            $string .= "<errors>";
+                foreach((array) $errors as $result){
+                    if (!$result->isSuccessful)
+                        $string .= "<error id=''>" . $result->message . "</error>";
+                }
+            $string .= "</errors>";
+
+            }
+            else {
+            $string .= "<ok>All operations successful</ok>";
```

Comment:
I think you should let code flow into the `if (!isset($string))` block below in case of success. One of the things that block does is to return the data of the created projects including their newly assigned IDs, which the front-end can use to issue update operations later.

It's unclear what to do if some projects fail and some don't, but we'll figure out if we ever use the feature...

Reply:
This is still not working as expected. The service returns this in case of success creating one project:
```
<?xml version="1.0"?>
<return service="createProjects"><error id="1">Project created successfully.</error></return>
```

And this in case of multiple projects:
```
<?xml version="1.0"?>
<return service="createProjects"/>
```

Reply:
Looks like my xml was a bit malformed (and formed at the wrong time).